pkg/kubectl/cmd/util/diff/diffoptions.go

```diff
@@ -121,6 +128,11 @@ func (o *DiffOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
 	return err
 }
 
+func deleteEmptyAnnotations(obj runtime.Object) {
+	delete(obj.(runtime.Unstructured).UnstructuredContent()["metadata"].(map[string]interface{}), "annotations")
```

Comment:
I made a little change, but do you mean directly use `SetAnnotation()` ? that looks good, but unfortunately, it doesn't works as you expected, still not remove annotation field, just set to empty, you can try it yourself.